### ISSUE TITLE
Failure to clone repository when `git` CLI succeeds

### ISSUE THREAD

**Issue by alexcrichton** [libgit2/libgit2#5525]
### Reproduction steps

Build and run this program:

```c
#include <assert.h>
#include <git2/clone.h>
#include <git2/global.h>
#include <git2/errors.h>
#include <stdio.h>

int main() {
  assert(git_libgit2_init() >= 0);
  git_repository *repo;
  const char *url = "https://chromium.googlesource.com/chromium/llvm-project/llvm/lib/Fuzzer";
  const char *local = "local-dir";
  int rc = git_clone(&repo, url, local, NULL);
  if (rc == GIT_OK) {
    printf("ok!\n");
    return 0;
  }
  const git_error *err = git_error_last();
  printf("rc %d\n", rc);
  printf("error %s\n", err->message);
  printf("klass %d\n", err->klass);
}
```

compared to:

```
$ git clone https://chromium.googlesource.com/chromium/llvm-project/llvm/lib/Fuzzer
Cloning into 'Fuzzer'...
remote: Total 4444 (delta 3448), reused 4444 (delta 3448)
Receiving objects: 100% (4444/4444), 669.15 KiB | 3.05 MiB/s, done.
Resolving deltas: 100% (3448/3448), done.
$ echo $?
0
```

### Expected behavior

The same result of cloning for libgit2 vs the `git` CLI

### Actual behavior

It looks like libgit2 fails with an inflate error, but the `git` CLI succeeds

### Version of libgit2 (release number or SHA1)

b7b872f51074ef8231ae8b94dd93eada127c9365 (current master)

### Operating system(s) tested

Linux (Ubuntu 18.04)

---

This was originally reported at https://github.com/rust-lang/cargo/issues/8258 and I'm forwarding here because I'm not sure how best to next debug this. Is there a way to extract out the stream of data to inspect whether it's actually an invalid zlib stream? Is this perhaps a case where the git CLI isn't checking an error code?

Some debugging showed that the inflation call [failed here](https://github.com/madler/zlib/blob/cacf7f1d4e3d44d871b605da3b647f07d718623f/inflate.c#L1212-L1213) which seems like it's a maybe-optional check (that seems like a checksum?) which the `git` CLI may be ignoring.

I'm mostly curious if libgit2 folks can help debug this to see if it's an issue with this library, the server, or the `git` CLI

**Comment by ethomson** [comment]
🤔  There's no trivial way to filter out the stream directly, but we could probably figure out a way to add some hooks.  (At a bare minimum, we can produce a hacked up branch that dumps it to a temp file.)

I'm not sure when I can attack this problem though.  😬

**Comment by paolobarbolini** [comment]
I tried running the clone example on multiple commits and I think this was introduced in #5286.

Last commit that worked: https://github.com/libgit2/libgit2/commit/6c21c989a3b8cba20ad8cbe806937071a1923832
First commit I was able to test that broke it: https://github.com/libgit2/libgit2/commit/b9c5b15a7958ab4ecb83504a6d858efd18610297

I'm not familiar with this project or C in general but I hope this helps.

**Comment by ethomson** [comment]
I don't have a solution yet, but I can reproduce this problem locally.

**Comment by ethomson** [comment]
@paolobarbolini and @alexcrichton would you be able to sanity-check me on #5536 ?

**Comment by ethomson** [comment]
I released [v1.0.1](https://github.com/libgit2/libgit2/releases/tag/v1.0.1) with a fix for this issue.
